crux-core/test/crux/datascript_pull_test.clj

```diff
@@ -125,37 +78,40 @@
              (crux/q db '{:find [(eql/project ?e [* {:parent [:crux.db/id]}])]
                           :where [[?e :crux.db/id :david]]})))))
 
-;; TODO `:limit`
-#_
 (t/deftest test-pull-limit
-  (let [db (d/init-db
-            (concat
-             test-datoms
-             [(d/datom 4 :friend 5)
-              (d/datom 4 :friend 6)
-              (d/datom 4 :friend 7)
-              (d/datom 4 :friend 8)]
-             (for [idx (range 2000)]
-               (d/datom 8 :aka (str "aka-" idx))))
-            test-schema)]
+  (let [db (submit-test-docs (concat people-docs
+                                     [{:crux.db/id :elizabeth,
+                                       :name "Elizabeth"
+                                       :friend #{:matthew :eunan :kerri :rebecca}
+                                       :aka (into #{} (map #(str "liz-" %)) (range 2000))}]))]
 
+    ;; I'm tempted not to have a default - might be a gotcha? RFC?
```

Comment:
Do we want to resolve this RFC before merging? Or OK to follow it up after?

Reply:
@jonpither yep, happy to resolve it now - wdyt?

Reply:
Assuming we did opt for a similar default limit mechanism, presumably it shouldn't be as high as 1000 based on the numbers in https://github.com/juxt/crux/issues/1353

I'm also inclined to think that a _silently_ default limit isn't a good idea as it could cause subtle errors. Short of implementing a more stack-friendly alternative approach, I think it would be better that a limit is always enforced as as manually user-specified for all recursive usage.

Reply:
@refset this limit thankfully doesn't affect the stack depth. the one in #1353 is related to the nesting of the query (vertical depth of the tree), if you like - this is elements in a collection at one level (horizontal/breadth).

Reply:
propose that, given we don't have a limit by default on the main query, we don't limit here?